src/metabase/automagic_dashboards/core.clj

```diff
@@ -496,25 +504,28 @@
         subform))
     {:type     :query
      :database (-> context :root :database)
-     :query    (cond-> {:source_table (if (->> context :source (instance? (type Table)))
+     :query    (cond-> {:source-table (if (->> context :source (instance? (type Table)))
                                         (-> context :source u/get-id)
                                         (->> context :source u/get-id (str "card__")))}
-                 (not-empty filters)
-                 (assoc :filter (->> filters
-                                     (map :filter)
-                                     (apply qp.expand/merge-filter-clauses)))
-
-                 (not-empty dimensions)
+                 (seq filters)
+                 (assoc :filter (vec
+                                 (cons
+                                  :and
+                                  (map (comp (partial normalize/normalize-fragment [:query :filter])
+                                             :filter)
+                                       filters))))
+
+                 (seq dimensions)
                  (assoc :breakout dimensions)
 
-                 (not-empty metrics)
```

Comment:
is there a reason beyond aesthetic preference to prefer seq to not-empty?

Reply:
Just aesthetic. Sorry for changing this. Sometimes I get carried away changing everything. I'll change it back

Reply:
No worries. I'm happy if we make a coding style decision and unify such things. Clearly I'm in favour of not-empty as it's more semantic, but I'm even more in favour of a unified style.

Reply:
Yeah I was going by this 

https://github.com/bbatsov/clojure-style-guide#nil-punning